app/src/main/java/com/infomaniak/swisstransfer/ui/screen/newtransfer/TransferSendManager.kt

```diff
@@ -53,58 +53,102 @@ class TransferSendManager @Inject constructor(
     private val _integrityCheckResult = MutableStateFlow(AppIntegrityResult.Idle)
     val integrityCheckResult = _integrityCheckResult.asStateFlow()
 
-    //region App Integrity
     suspend fun sendTransfer(newUploadSession: NewUploadSession) {
         _integrityCheckResult.value = AppIntegrityResult.Ongoing
-        coroutineScope {
-            runCatching {
-                appIntegrityManager.getChallenge(
-                    onSuccess = { requestAppIntegrityToken(appIntegrityManager, newUploadSession) },
-                    onFailure = ::setFailedIntegrityResult,
-                )
-            }.onFailure { exception ->
-                SentryLog.e(TAG, "Failed to start the upload", exception)
+
+        withIntegrityToken(
+            onSuccess = { attestationToken ->
+                sendTransfer(newUploadSession, attestationToken)
+            },
+            onRefused = {
+                _integrityCheckResult.value = AppIntegrityResult.Fail
+            },
+            onFailure = { exception ->
+                if (exception !is CancellationException) {
+                    SentryLog.e(TAG, "Integrity token received an exception", exception)
+                } else {
+                    SentryLog.i(TAG, "Integrity token received an exception", exception)
+                }
                 _sendActionResult.update { SendActionResult.Failure }
             }
```

Comment:
This onFailure seems useless as it won't catch exceptions happening inside coroutines, and then sendTransfer has its own runCatching
But better safe than sorry I guess, but at least we could directly put this in the onFailure in the function and not expose both onFailure and onRefused(), while onRefused is already applicated when an exeception has been thrown somewhere during appIntegrity

Reply:
We'll fix this in the next PR that already modify this part